### PR TITLE
graph-data.rs: verify channel order

### PR DESCRIPTION
**Pull request by vrutkovs**
This adds a CI tests which ensures, that releases are properly ordered: first it must be placed in candidate, then fast and then stable. If a release is put in candidate and then in stable, it would throw an error.

EUS channel is omitted for now, as its not yet entirely clear how releases would be structured there.

https://issues.redhat.com/browse/OTA-126

/cc @LalatenduMohanty @wking @PratikMahajan

### CODE REVIEW

File: graph-data.rs/src/check_channels.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    for r in releases {
+        let release_vec: Vec<String> = CHANNEL_ORDER
+            .iter()
+            .map(|v| format!("{}-{}", v, r))
+            .collect();
+        // Iterate over pairs
+        for (a, b) in release_vec.iter().tuple_windows() {
+            // Check that channels are present
+            if channels.get(a).is_none() || channels.get(b).is_none() {
+                continue;
```
Comment (wking) [inline graph-data.rs/src/check_channels.rs:37]: If `a` is present, it is an error for `b` to be absent (e.g. non-empty stable-4.6 requires fast 4.6 to exist).

File: graph-data.rs/src/check_channels.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    // Collect a list of releases (4.6, 4.5 etc.)
+    let releases: Vec<&str> = channels_vec
+        .iter()
+        .filter_map(|c| c.name.split("-").last())
+        .unique()
+        .collect();
+
+    // Prepare a list of expected channels, ordered according to the rule
+    for r in releases {
+        let release_vec: Vec<String> = CHANNEL_ORDER
```
Comment (wking) [inline graph-data.rs/src/check_channels.rs:29]: Special-case 4.1, where we had `presubmit-4.1` and `stable-4.1`?

File: graph-data.rs/src/check_channels.rs (hunk 1 above)
Comment (vrutkovs) [reply graph-data.rs/src/check_channels.rs:37]: This check makes sense, but it currently fails: `Channel stable-4.1 exists, but not fast-4.1`.

I wonder if I should :
* add `fast-4.1` and `candidate-4.1`
* treat "prerelease" as "fast and candidate"
* or ignore 4.1 all together

(picking the latter for now)

File: graph-data.rs/src/check_channels.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            errors.push(e.unwrap())
+        }
+    }
+    if errors.is_empty() {
+        Ok(())
+    } else {
+        Err(anyhow::anyhow!("Error in channel ordering: {:?}", errors))
+    }
+}
```
Comment (LalatenduMohanty) [inline graph-data.rs/src/check_channels.rs:75]: @vrutkovs Now that we have some sizable code here we should add unit tests. It does not have to go in this PR but we should add unit tests soon.

File: graph-data.rs/src/check_channels.rs (hunk 2 above)
Comment (LalatenduMohanty) [reply graph-data.rs/src/check_channels.rs:55]: I guess so. @vrutkovs can confirm.

File: graph-data.rs/src/check_channels.rs (hunk 3 above)
Comment (vrutkovs) [reply graph-data.rs/src/check_channels.rs:75]: Good idea, filed OTA-342

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**